Add cookies section on the SDK spec terminology (#1346)
clerk/clerk-docs: docs/references/sdk/terminology.mdx

| Authorization header verification | This refers to the HTTP Authorization request header. Clerk uses this HTTP header instead of the `__client` cookie for secure communication with the FAPI in non-standard web browser environments. For example in React Native apps, Chrome extensions, and hybrid apps like Electron. The `__client` JWT will be stored in a secure storage provided by the platform and injected into the HTTP header of each FAPI request. |
| Handshake | The client handshake is a mechanism that is used to resolve a request’s authentication state from “unknown” to definitively signed in or signed out. Clerk’s session management architecture relies on a short-lived session JWT to validate requests, along with a long-lived session that is used to keep the session JWT fresh by interacting with the FAPI. The long-lived session token is stored in an `HttpOnly` cookie associated with the FAPI domain. If a short-lived session JWT is expired on a request to an application’s backend, the SDK doesn’t know if the session has ended, or if a new short-lived JWT needs to be issued. When an SDK gets into this state, it triggers the handshake. |

## Cookies

> Clerk is using cookies in order to keep user sessions alive. The most important cookies are the following:

Reviewer: ```suggestion
Clerk is using cookies in order to keep user sessions alive. It's useful to know about these when debugging issues.
```

I think by putting the table below we implicitly say that these are the most important ones, so we can skip that sentence

### Development instance cookies

| Cookie Name | Description |
| - | - |

> | `__clerk_db_jwt` | This is called `Dev Browser JWT` and helps Clerk sync the session between different domains, because in the dev environment the app is running on `localhost` and Clerk FAPI and Account Portal runs under the `accounts.dev` domain. |

Reviewer: ```suggestion
| `__clerk_db_jwt` | When using the Development instance, the **Development Browser JWT** helps Clerk sync the session between different domains (`localhost` and `https://name.accounts.dev`) |
```

We mentioned "Instances" above, so keeping this in sync here. Also reworded a bit to make it more concise

Reviewer 2: ```suggestion
| `__clerk_db_jwt` | The **Development Browser JWT**. Helps Clerk sync the session between the development instance domain, `localhost`, and the Clerk FAPI and Account Portal domain, `https://name.accounts.dev`. |
```

> | `__client_uat` | Its naming is coming from `Client Updated At` and it's a hint for Clerk to check if the user session is still valid. It is set by the Clerk.js on the client side. |

Reviewer: ```suggestion
| `__client_uat` | `uat` stands for **Client Updated At** and helps Clerk to check if the user session is still valid. It is set by clerk-js on the client-side. |
```

We mentioned clerk-js above in the table, so aligning it here

Reviewer 2: ```suggestion
| `__client_uat` | `uat` stands for **Client Updated At** and helps Clerk to check if the user session is still valid. It is set by `clerk-js` on the client-side. |
```

Author: Maybe this?
```suggestion
| `__client_uat` | `uat` stands for **Updated At** and helps Clerk to check if the user session is still valid. It is set by `clerk-js` on the client-side. |
```

Reviewer 2: looks good!

> | `__session` | A short-lived session JWT to validate request in your application or your API. |

Reviewer 2: ```suggestion
| `__session` | A short-lived session JWT to validate requests in your application or your API. |
```

### Production instance cookies

| Cookie Name | Description |
| - | - |

> | `__client` | This is a long-lived HttpOnly cookie that is used to issue short-lived session JWTs. It is set by the Clerk FAPI. |

Reviewer: ```suggestion
| `__client` | This is a long-lived `HttpOnly` cookie that is used to issue short-lived session JWTs. It is set by the Clerk FAPI. |
```

> | `__client_uat` | As described above, but in Production this is set by the Clerk FAPI. |

Reviewer 2: ```suggestion
| `__client_uat` | `uat` stands for **Client Updated At** and helps Clerk to check if the user session is still valid. It is set by the Clerk FAPI. |
```

Author: Maybe this?
```suggestion
| `__client_uat` | `uat` stands for **Updated At** and helps Clerk to check if the user session is still valid. It is set by the Clerk FAPI. |
```

Reviewer 2: looks good!

> | `__session` | A short-lived session JWT to validate request in your application or your API. |

Reviewer 2: ```suggestion
| `__session` | A short-lived session JWT to validate requests in your application or your API. |
```

> Clerk is using more non-persistent cookies for internal mechanisms (e.g. `__clerk_handshake`).

Reviewer: ```suggestion
Clerk is using other non-persistent cookies (e.g. `__clerk_handshake`) for internal mechanisms but for the sake of creating a new SDK from scratch you don't need to worry about those.
```
